livy/src/main/java/org/apache/zeppelin/livy/BaseLivyInterpreter.java

```diff
@@ -331,7 +331,13 @@ private InterpreterResult appendSessionExpire(InterpreterResult result, boolean
   private InterpreterResult getResultFromStatementInfo(StatementInfo stmtInfo,
                                                        boolean displayAppInfo) {
     if (stmtInfo.output != null && stmtInfo.output.isError()) {
-      return new InterpreterResult(InterpreterResult.Code.ERROR, stmtInfo.output.evalue);
+      InterpreterResult result = new InterpreterResult(InterpreterResult.Code.ERROR);
+      result.add(stmtInfo.output.evalue);
+      if (stmtInfo.output.traceback != null) {
+        for (int i = 0; i < stmtInfo.output.traceback.length; i++)
+          result.add(stmtInfo.output.traceback[i]);
```

Comment:
What do you mean by splitting?  By adding traceback into into result, Zeppelin can display more info about the error Livy server encounter when executing spark code. I have put two more screen shot when using Livy 0.4 to run spark streaming job. One with traceback added to result, one without.